components/homme/src/theta-l/model_init_mod.F90

```diff
@@ -81,23 +81,32 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
        ! sponge layer starts at p=4*ptop 
        ! 
        ! some test cases have ptop=200mb
-       ptop_over_press = hvcoord%etai(1) / hvcoord%etam(k)  ! pure sigma coordinates has etai(1)=0
-
-       ! active for p<4*ptop (following cd_core.F90 in CAM-FV)
-       ! CAM 26L and 30L:  top 2 levels
-       ! E3SM 72L:  top 4 levels
+       if (hvcoord%etai(1)==0) then
+          ! pure sigma coordinates could have etai(1)=0
+          ptop_over_press = hvcoord%etam(1) / hvcoord%etam(k)  
```

Comment:
for pure sigma A coefficients are all zero, or not?

Reply:
some of our pure sigma coordinate files have A(1)=0  (top of model is at p=0)

I've never seen this done with hybrid coordinates, but that seems to be a convention - nothing would prevent it.